sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -185,19 +190,21 @@ public Builder withValidation() {
   public static class Builder {
     private final String defaultAppName;
     private final String[] args;
+    private final Map<String, String> options;
```

Comment:
Hmm, it is not about converting a command line to a map but taking a map as input to build the map<String, Object> (the generics are important here) which are the backing instances of the proxy.

Both are merged in a single source which is then converted in instances used by the proxy.
The internal format is a multimap but I don't want to put that in an API so the Map<String, String> is at the exact same level than the String[] IMHO, no?

Reply:
Command line is a `Map<String, String>` so that's the common denominator. I know properties support something like `Map<String, Object>` but for properties files and `-D` it is again a `Map<String, String>`. The conversion from `Map<String, String>` to `Map<String, Object>` is what `parseObjects` does.

Reply:
-D is not Map<String, String> but a multimap since you can pass N times the same option so we dont have a common structure except Map<String, Object>. Will try to move to that as an internal.